sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java

```diff
@@ -64,12 +67,16 @@ class StorageAccountImpl
     private StorageAccountUpdateParameters updateParameters;
     private StorageNetworkRulesHelper networkRulesHelper;
     private StorageEncryptionHelper encryptionHelper;
+    private StorageAccountMsiHandler storageAccountMsiHandler;
+    private final AuthorizationManager authorizationManager;
 
-    StorageAccountImpl(String name, StorageAccountInner innerModel, final StorageManager storageManager) {
+    StorageAccountImpl(String name, StorageAccountInner innerModel, final StorageManager storageManager, final AuthorizationManager authorizationManager) {
         super(name, innerModel, storageManager);
+        this.authorizationManager = authorizationManager;
         this.createParameters = new StorageAccountCreateParameters();
         this.networkRulesHelper = new StorageNetworkRulesHelper(this.createParameters);
         this.encryptionHelper = new StorageEncryptionHelper(this.createParameters);
+        this.storageAccountMsiHandler = new StorageAccountMsiHandler(authorizationManager, this).withCreateParameters(createParameters);
```

Comment:
Could you try making createParameters and updateParameters package-private in `StorageAccountImpl`? 
This way, you can refer them by `storageAccount.createParameters` and `storageAccount.updateParameters` in `MsiHandler`, and no need for `withCreateParameters` and `withUpdateParameters`.

Reply:
Fixed in the new version.